internal/k8s/validator.go

```diff
@@ -371,7 +372,25 @@ func (l *leader) ValidateAviInfraSetting(key string, infraSetting *akov1beta1.Av
 			refData[nodeNetwork.NetworkName] = "Network"
 		}
 	}
+	tenant := lib.GetTenant()
+	var err error
 	if infraSetting.Spec.SeGroup.Name != "" {
+		// In case of WCP/VCF deployments in VPC mode, NS can have SeGroup(shared/dedicated)
+		// annotation which results in setting SeGroup in infrasetting and for validation of SeGroup,
+		// tenant needs to be determined through vpc path in VPCCR.
+		if lib.GetVPCMode() && infraSetting.Spec.NSXSettings.T1LR != nil {
+			aviClientPool := cache.SharedAVIClients(lib.GetTenant())
+			if aviClientPool == nil || len(aviClientPool.AviClient) == 0 {
+				return fmt.Errorf("avi rest client initialization failed")
+			}
+			vpcArr := strings.Split(*infraSetting.Spec.NSXSettings.T1LR, "/vpcs/")
+			projectArr := strings.Split(vpcArr[0], "/projects/")
+			tenant, err = lib.GetTenantForProject(projectArr[len(projectArr)-1], aviClientPool.AviClient[0])
+			if err != nil {
+				utils.AviLog.Warnf("failed to fetch admin tenant from Avi, error: %s", err.Error())
```

Comment:
Let's following the existing logging model by adding Key to the log message. Also, need to update the AviInfraSetting status to Rejected as we are returning from here.